src/Products/Viewer/Controllers/ViewerApiController.cs

```diff
@@ -438,31 +416,36 @@ private static PageDescriptionEntity GetPageInfo(Page page, string pagesInfoPath
         /// <param name="documentGuid">Document guid.</param>
         /// <param name="settings">Viewer settings object.</param>
         /// <returns>Page content as a string.</returns>
-        private static string GetPageContent(Page page, string password, string documentGuid, ViewerSettings settings)
+        private static string GetPageContent(int pageNumber, string password, string documentGuid, string cachePath, int passedPageNumber = -1, int newAngle = 0)
         {
-            List<MemoryStream> pages = new List<MemoryStream>();
+            var fileFolderName = Path.GetFileName(documentGuid).Replace(".", "_");
 
-            using (GroupDocs.Viewer.Viewer viewer = new GroupDocs.Viewer.Viewer(documentGuid, GetLoadOptions(password), settings))
+            if (globalConfiguration.Viewer.GetIsHtmlMode())
             {
-                MemoryPageStreamFactory pageStreamFactory = new MemoryPageStreamFactory(pages);
-
-                if (globalConfiguration.Viewer.GetIsHtmlMode())
-                {
-                    ViewOptions viewOptions = HtmlViewOptions.ForEmbeddedResources(pageStreamFactory);
-                    viewOptions.SpreadsheetOptions.TextOverflowMode = TextOverflowMode.HideText;
+                string filePath = $"{cachePath}/{fileFolderName}/p{pageNumber}.html";
+                return File.ReadAllText(filePath);
+            }
+            else
+            {
+                //if (passedPageNumber >= 0 && newAngle != 0)
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/groupdocs-viewer/GroupDocs.Viewer-for-.NET-MVC/pullRequest?prid=5933705)